--- a/scripts/package/builddeb	2020-04-17 06:04:18.000000000 -0400
+++ b/scripts/package/builddeb	2020-04-17 08:00:14.095318486 -0400
@@ -92,6 +92,7 @@
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	cp arch/arm64/boot/Image "$tmpdir/boot/kernel8.img"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
 
@@ -139,6 +140,12 @@
 	mv $libc_headers_dir/usr/include/asm $libc_headers_dir/usr/include/$host_arch/
 fi
 
+# Install dtb and overlays
+mkdir -p "$tmpdir/boot/broadcom/overlays"
+cp -r "$tmpdir/usr/lib/linux-image-$version/broadcom/bcm2710-rpi-3-b-plus.dtb" "$tmpdir/boot/broadcom/bcm2710-rpi-3-b-plus.dtb"
+cp -r "$tmpdir/usr/lib/linux-image-$version/broadcom/bcm2710-rpi-3-b.dtb" "$tmpdir/boot/broadcom/bcm2710-rpi-3-b.dtb"
+cp -r "$tmpdir/usr/lib/linux-image-$version/overlays/" "$tmpdir/boot/broadcom/"
+
 # Install the maintainer scripts
 # Note: hook scripts under /etc/kernel are also executed by official Debian
 # kernel packages, as well as kernel packages built using make-kpkg.
