diff -Naur a/scripts/package/builddeb b/scripts/package/builddeb
--- a/scripts/package/builddeb	2021-12-07 19:34:46.734175000 -0500
+++ b/scripts/package/builddeb	2022-01-18 14:47:33.199644114 -0500
@@ -51,8 +51,8 @@
 kernel_headers_dir="$objtree/debian/hdrtmp"
 libc_headers_dir="$objtree/debian/headertmp"
 dbg_dir="$objtree/debian/dbgtmp"
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 libc_headers_packagename=linux-libc-dev
 dbg_packagename=$packagename-dbg
 
@@ -92,6 +92,16 @@
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	if [ -e arch/arm/boot/zImage ]; then
+		cp arch/arm/boot/zImage "$tmpdir/boot/kernel7l.img";
+	fi
+	if [ -e arch/arm64/boot/Image ]; then
+		cp arch/arm64/boot/Image "$tmpdir/boot/kernel8.img";
+	fi
+	install -m 0644 firmware/fixup4*.dat "$tmpdir/boot"
+	install -m 0644 firmware/start4*.elf "$tmpdir/boot"
+	cp firmware/LICENCE.broadcom "$tmpdir/boot"
+	cp firmware/COPYING.linux "$tmpdir/boot"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
 
@@ -139,6 +149,40 @@
 	mv $libc_headers_dir/usr/include/asm $libc_headers_dir/usr/include/$host_arch/
 fi
 
+# install dtb and overlays
+if [ -e $tmpdir/boot ]; then :; else mkdir -p "$tmpdir/boot"; fi
+# armhf
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-4-b.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-4-b.dtb" "$tmpdir/boot/bcm2711-rpi-4-b.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-400.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-400.dtb" "$tmpdir/boot/bcm2711-rpi-400.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-cm4.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-cm4.dtb" "$tmpdir/boot/bcm2711-rpi-cm4.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-cm4s.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2711-rpi-cm4s.dtb" "$tmpdir/boot/bcm2711-rpi-cm4s.dtb";
+fi
+# arm64
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-4-b.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-4-b.dtb" "$tmpdir/boot/bcm2711-rpi-4-b.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-400.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-400.dtb" "$tmpdir/boot/bcm2711-rpi-400.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-cm4.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-cm4.dtb" "$tmpdir/boot/bcm2711-rpi-cm4.dtb";
+fi
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-cm4s.dtb ]; then
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-cm4s.dtb" "$tmpdir/boot/bcm2711-rpi-cm4s.dtb";
+fi
+# overlays
+if [ -e $tmpdir/usr/lib/raspberrypi-linux-image/overlays ]; then
+	mkdir -p "$tmpdir/boot/overlays";
+	cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/overlays/" "$tmpdir/boot/";
+fi
+
 # Install the maintainer scripts
 # Note: hook scripts under /etc/kernel are also executed by official Debian
 # kernel packages, as well as kernel packages built using make-kpkg.
@@ -163,6 +207,21 @@
 EOF
 	chmod 755 "$tmpdir/DEBIAN/$script"
 done
+        cat > $tmpdir/DEBIAN/preinst <<EOT
+#!/bin/bash
+# purge kernel script
+
+set -e
+
+if [ -e /boot/kernel*.img ]; then
+	rm -fdr /boot/{kernel*,initrd.img*,System.map-*,config-*,vmlinuz-*,*.dtb,fixup4*.dat,start4*.elf,LICENCE.broadcom,COPYING.linux,overlays};
+else
+	echo -e "\e[1;37mInstalling new kernel\e[0m ...";
+fi
+
+exit 0
+EOT
+        chmod 755 $tmpdir/DEBIAN/preinst
 
 # Build kernel header package
 (cd $srctree; find . -name Makefile\* -o -name Kconfig\* -o -name \*.pl) > "$objtree/debian/hdrsrcfiles"
@@ -183,6 +242,52 @@
 (cd $objtree; cp $KCONFIG_CONFIG $destdir/.config) # copy .config manually to be where it's expected to be
 ln -sf "/usr/src/linux-headers-$version" "$kernel_headers_dir/lib/modules/$version/build"
 rm -f "$objtree/debian/hdrsrcfiles" "$objtree/debian/hdrobjfiles"
+if [ -e headers-byteshift.patch ]; then
+	cp headers-byteshift.patch $destdir/
+	cp $srctree/scripts/module.lds $destdir/
+	mkdir -p $kernel_headers_dir/DEBIAN
+	cat > $kernel_headers_dir/DEBIAN/postinst <<EOT
+#!/bin/bash
+# compile headers
+
+set -e
+
+clean_headers(){
+find -type f -exec touch {} +
+if grep -w "Linux/arm" ".config"; then
+	echo 'y' | make ARCH=arm M=scripts clean;
+else
+	echo 'y' | make M=scripts clean;
+fi
+patch -p1 < headers-byteshift.patch
+if [ -e scripts/module.lds ]; then :; else install -m 0644 module.lds scripts/; fi
+rm -f {headers-byteshift.patch,module.lds}
+if grep -w "Linux/arm" ".config"; then
+	echo 'y' | make ARCH=arm -j\$(grep -c 'processor' /proc/cpuinfo) -s scripts;
+	echo 'y' | make ARCH=arm -j\$(grep -c 'processor' /proc/cpuinfo) -s M=scripts/mod/;
+else
+	echo 'y' | make -j\$(grep -c 'processor' /proc/cpuinfo) -s scripts;
+	echo 'y' | make -j\$(grep -c 'processor' /proc/cpuinfo) -s M=scripts/mod/;
+fi
+if [ -e .config.old ]; then rm -f .config.old; fi
+}
+
+cd /usr/src/linux-headers-$version
+echo -e "\e[1;37mCompiling headers\e[0m ..."
+clean_headers > /dev/null 2>&1
+exit 0
+EOT
+	chmod 755 $kernel_headers_dir/DEBIAN/postinst
+fi
+	mkdir -p $kernel_headers_dir/DEBIAN; cat > $kernel_headers_dir/DEBIAN/preinst <<EOT
+#!/bin/sh
+# remove old linux headers
+
+set -e
+rm -fdr /usr/src/linux-headers-* > /dev/null 2>&1
+exit 0
+EOT
+        chmod 755 $kernel_headers_dir/DEBIAN/preinst
 
 if [ "$ARCH" != "um" ]; then
 	create_package "$kernel_headers_packagename" "$kernel_headers_dir"
diff -Naur a/scripts/package/mkdebian b/scripts/package/mkdebian
--- a/scripts/package/mkdebian	2021-12-07 19:34:46.735175000 -0500
+++ b/scripts/package/mkdebian	2022-01-18 14:47:10.425112028 -0500
@@ -94,8 +94,8 @@
 	packageversion=$version-$revision
 fi
 sourcename=$KDEB_SOURCENAME
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 dbg_packagename=$packagename-dbg
 debarch=
 set_debarch
