diff -Naur a/scripts/package/builddeb b/scripts/package/builddeb
--- a/scripts/package/builddeb	2020-06-10 05:54:19.000000000 -0400
+++ b/scripts/package/builddeb	2020-06-10 08:56:13.240213745 -0400
@@ -47,12 +47,13 @@
 }
 
 version=$KERNELRELEASE
+kernel_date=`date +%F`
 tmpdir="$objtree/debian/tmp"
 kernel_headers_dir="$objtree/debian/hdrtmp"
 libc_headers_dir="$objtree/debian/headertmp"
 dbg_dir="$objtree/debian/dbgtmp"
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 libc_headers_packagename=linux-libc-dev
 dbg_packagename=$packagename-dbg
 
@@ -92,6 +93,7 @@
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	cp arch/arm/boot/zImage "$tmpdir/boot/kernel.img"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
 
@@ -139,6 +141,14 @@
 	mv $libc_headers_dir/usr/include/asm $libc_headers_dir/usr/include/$host_arch/
 fi
 
+# install dtb and overlays
+mkdir -p "$tmpdir/boot/overlays"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2708-rpi-zero-w.dtb" "$tmpdir/boot/bcm2708-rpi-zero-w.dtb"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2708-rpi-zero.dtb" "$tmpdir/boot/bcm2708-rpi-zero.dtb"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2708-rpi-b.dtb" "$tmpdir/boot/bcm2708-rpi-b.dtb"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/bcm2708-rpi-b-plus.dtb" "$tmpdir/boot/bcm2708-rpi-b-plus.dtb"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/overlays/" "$tmpdir/boot/"
+
 # Install the maintainer scripts
 # Note: hook scripts under /etc/kernel are also executed by official Debian
 # kernel packages, as well as kernel packages built using make-kpkg.
@@ -164,6 +174,40 @@
 	chmod 755 "$tmpdir/DEBIAN/$script"
 done
 
+       # remove kernel
+        cat > $tmpdir/DEBIAN/preinst <<EOT
+#!/bin/sh
+# purge kernel script
+
+set -e
+
+if touch -c /boot/overlays > /dev/null 2>&1; then rm -fdr /boot/overlays;
+        else echo "Missing overlays directory." &>/dev/null;
+fi
+
+if ls /boot/*.dtb > /dev/null 2>&1; then rm -fr /boot/*.dtb;
+        else echo "Missing dtb files." &>/dev/null;
+fi
+
+if ls /boot/System.map-* > /dev/null 2>&1; then rm -f /boot/System.map-*;
+        else echo "Missing System.map file." &>/dev/null;
+fi
+
+if ls /boot/config-* > /dev/null 2>&1; then rm -f /boot/config-*;
+        else echo "Missing config file." &>/dev/null;
+fi
+
+if ls /boot/vmlinuz-* > /dev/null 2>&1; then rm -f /boot/vmlinuz-*;
+        else echo "Missing vmlinuz file." &>/dev/null;
+fi
+
+if ls /boot/kernel* > /dev/null 2>&1; then rm -f /boot/kernel*;
+        else echo "Installing new kernel." &>/dev/null;
+fi
+exit 0
+EOT
+        chmod 755 $tmpdir/DEBIAN/preinst
+
 # Build kernel header package
 (cd $srctree; find . -name Makefile\* -o -name Kconfig\* -o -name \*.pl) > "$objtree/debian/hdrsrcfiles"
 (cd $srctree; find arch/*/include include scripts -type f -o -type l) >> "$objtree/debian/hdrsrcfiles"
@@ -183,6 +227,37 @@
 (cd $objtree; cp $KCONFIG_CONFIG $destdir/.config) # copy .config manually to be where it's expected to be
 ln -sf "/usr/src/linux-headers-$version" "$kernel_headers_dir/lib/modules/$version/build"
 rm -f "$objtree/debian/hdrsrcfiles" "$objtree/debian/hdrobjfiles"
+cp headers-byteshift.patch $destdir/
+(cd $destdir; make M=scripts clean) && (cd $destdir; patch -p1 < headers-byteshift.patch)
+rm -f $destdir/headers-byteshift.patch
+
+	# make scripts
+	mkdir -p $kernel_headers_dir/DEBIAN
+	cat > $kernel_headers_dir/DEBIAN/postinst <<EOT
+#!/bin/sh
+# compile headers
+
+set -e
+cd /usr/src/linux-headers-$version
+echo "Compiling headers ..."
+find -type f -exec touch {} +
+make -s scripts >/dev/null
+make -s M=scripts/mod/ >/dev/null
+exit 0
+EOT
+
+        chmod 755 $kernel_headers_dir/DEBIAN/postinst
+
+       # remove old linux headers 
+        cat > $kernel_headers_dir/DEBIAN/preinst <<EOT
+#!/bin/sh
+# remove old linux headers
+
+set -e
+rm -fdr /usr/src/linux-headers-* > /dev/null 2>&1
+exit 0
+EOT
+        chmod 755 $kernel_headers_dir/DEBIAN/preinst
 
 if [ "$ARCH" != "um" ]; then
 	create_package "$kernel_headers_packagename" "$kernel_headers_dir"
diff -Naur a/scripts/package/mkdebian b/scripts/package/mkdebian
--- a/scripts/package/mkdebian	2020-06-10 05:54:19.000000000 -0400
+++ b/scripts/package/mkdebian	2020-06-10 08:55:31.537376906 -0400
@@ -94,8 +94,8 @@
 	packageversion=$version-$revision
 fi
 sourcename=$KDEB_SOURCENAME
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 dbg_packagename=$packagename-dbg
 debarch=
 set_debarch
