--- a/makefiles/cmake/vmcs.cmake	2021-06-26 06:42:32.371574862 -0400
+++ b/makefiles/cmake/vmcs.cmake	2021-06-26 06:44:21.462480302 -0400
@@ -59,8 +59,8 @@
 #        PERMISSIONS OWNER_WRITE WORLD_READ)
 
 # provide hello_pi demos
-install(DIRECTORY host_applications/linux/apps/hello_pi
-        DESTINATION ${VMCS_INSTALL_PREFIX}/src)
+#install(DIRECTORY host_applications/linux/apps/hello_pi
+#        DESTINATION ${VMCS_INSTALL_PREFIX}/src)
 
 # provide header files
 #install(DIRECTORY host_applications/linux/libs/bcm_host/include
