diff -Naur a/makefiles/cmake/vmcs.cmake b/makefiles/cmake/vmcs.cmake
--- a/makefiles/cmake/vmcs.cmake	2023-04-21 15:51:40.011830638 -0400
+++ b/makefiles/cmake/vmcs.cmake	2023-04-21 15:53:51.268505442 -0400
@@ -59,8 +59,8 @@
 #        PERMISSIONS OWNER_WRITE WORLD_READ)
 
 # provide hello_pi demos
-install(DIRECTORY host_applications/linux/apps/hello_pi
-        DESTINATION ${VMCS_INSTALL_PREFIX}/src)
+#install(DIRECTORY host_applications/linux/apps/hello_pi
+#        DESTINATION ${VMCS_INSTALL_PREFIX}/src)
 
 # provide header files
 #install(DIRECTORY host_applications/linux/libs/bcm_host/include
