diff -Naur a/scripts/package/builddeb b/scripts/package/builddeb
--- a/scripts/package/builddeb	2021-06-15 06:22:38.904344000 -0400
+++ b/scripts/package/builddeb	2021-07-18 08:28:22.830070823 -0400
@@ -51,8 +51,8 @@
 kernel_headers_dir="$objtree/debian/hdrtmp"
 libc_headers_dir="$objtree/debian/headertmp"
 dbg_dir="$objtree/debian/dbgtmp"
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 libc_headers_packagename=linux-libc-dev
 dbg_packagename=$packagename-dbg
 
@@ -92,6 +92,11 @@
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	cp arch/arm64/boot/Image "$tmpdir/boot/kernel8.img"
+	install -m 0644 binaries/fixup4*.dat "$tmpdir/boot"
+	install -m 0644 binaries/start4*.elf "$tmpdir/boot"
+	cp binaries/LICENCE.broadcom "$tmpdir/boot"
+	cp binaries/COPYING.linux "$tmpdir/boot"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
 
@@ -139,6 +144,13 @@
 	mv $libc_headers_dir/usr/include/asm $libc_headers_dir/usr/include/$host_arch/
 fi
 
+# install dtb
+mkdir -p "$tmpdir/boot"
+cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-4-b.dtb" "$tmpdir/boot/bcm2711-rpi-4-b.dtb"
+if ls "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-400.dtb" > /dev/null 2>&1;
+	then cp -r "$tmpdir/usr/lib/raspberrypi-linux-image/broadcom/bcm2711-rpi-400.dtb" "$tmpdir/boot/bcm2711-rpi-400.dtb"
+fi
+
 # Install the maintainer scripts
 # Note: hook scripts under /etc/kernel are also executed by official Debian
 # kernel packages, as well as kernel packages built using make-kpkg.
@@ -164,6 +176,39 @@
 	chmod 755 "$tmpdir/DEBIAN/$script"
 done
 
+        cat > $tmpdir/DEBIAN/preinst <<EOT
+#!/bin/bash
+# purge kernel script
+
+set -e
+
+if ls /boot/*.dtb > /dev/null 2>&1;
+	then rm -fdr /boot/overlays /boot/*.dtb;
+fi
+
+if ls /boot/System.map-* > /dev/null 2>&1;
+	then rm -f /boot/System.map-*;
+fi
+
+if ls /boot/config-* > /dev/null 2>&1;
+	then rm -f /boot/config-*;
+fi
+
+if ls /boot/vmlinuz-* > /dev/null 2>&1;
+	then rm -f /boot/vmlinuz-*;
+fi
+
+if ls /boot/{fixup4*.dat,start4*.elf} > /dev/null 2>&1;
+	then rm -f /boot/{fixup4*.dat,start4*.elf,LICENCE.broadcom,COPYING.linux};
+fi
+
+if ls /boot/kernel* > /dev/null 2>&1; then rm -f /boot/kernel* /boot/initrd.img*;
+	else echo -e "\e[1;37mInstalling new kernel\e[0m ...";
+fi
+exit 0
+EOT
+        chmod 755 $tmpdir/DEBIAN/preinst
+
 # Build kernel header package
 (cd $srctree; find . -name Makefile\* -o -name Kconfig\* -o -name \*.pl) > "$objtree/debian/hdrsrcfiles"
 (cd $srctree; find arch/*/include include scripts -type f -o -type l) >> "$objtree/debian/hdrsrcfiles"
@@ -184,6 +229,17 @@
 ln -sf "/usr/src/linux-headers-$version" "$kernel_headers_dir/lib/modules/$version/build"
 rm -f "$objtree/debian/hdrsrcfiles" "$objtree/debian/hdrobjfiles"
 
+        mkdir -p $kernel_headers_dir/DEBIAN
+        cat > $kernel_headers_dir/DEBIAN/preinst <<EOT
+#!/bin/sh
+# remove old linux headers
+
+set -e
+rm -fdr /usr/src/linux-headers-* > /dev/null 2>&1
+exit 0
+EOT
+        chmod 755 $kernel_headers_dir/DEBIAN/preinst
+
 if [ "$ARCH" != "um" ]; then
 	create_package "$kernel_headers_packagename" "$kernel_headers_dir"
 	create_package "$libc_headers_packagename" "$libc_headers_dir"
diff -Naur a/scripts/package/mkdebian b/scripts/package/mkdebian
--- a/scripts/package/mkdebian	2021-06-15 06:22:38.904344000 -0400
+++ b/scripts/package/mkdebian	2021-07-18 08:27:12.658723929 -0400
@@ -94,8 +94,8 @@
 	packageversion=$version-$revision
 fi
 sourcename=$KDEB_SOURCENAME
-packagename=linux-image-$version
-kernel_headers_packagename=linux-headers-$version
+packagename=raspberrypi-linux-image
+kernel_headers_packagename=raspberrypi-linux-headers
 dbg_packagename=$packagename-dbg
 debarch=
 set_debarch
