--- a/debian/control	2021-07-10 08:18:37.352760542 -0400
+++ b/debian/control	2021-07-10 08:20:41.335787659 -0400
@@ -10,8 +10,7 @@
 
 Package: raspberrypi-sys-mods
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, libcap2-bin, systemd (>= 230),
- gettext-base, procps
+Depends: ${shlibs:Depends}, ${misc:Depends}, libcap2-bin, gettext-base, procps
 Breaks: raspberrypi-ui-mods (<<1.20210706)
 Replaces: raspberrypi-ui-mods (<<1.20210706)
 Recommends: rfkill, rpi-eeprom
