--- a/debian/control	2020-06-12 13:50:07.005489917 -0400
+++ b/debian/control	2020-06-12 13:49:43.062442497 -0400
@@ -10,8 +10,7 @@
 
 Package: raspberrypi-sys-mods
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, libcap2-bin, systemd (>= 230),
- gettext-base
+Depends: ${shlibs:Depends}, ${misc:Depends}, libcap2-bin, gettext-base
 Recommends: rfkill, rpi-eeprom
 Description: System tweaks for the Raspberry Pi
  Various modifications to improve the performance or user experience.
