--- a/debian/rules	2021-11-22 09:02:59.417245204 -0500
+++ b/debian/rules	2021-11-22 09:05:43.660508613 -0500
@@ -34,16 +34,3 @@
 	dh_installinit --name=apply_noobs_os_config --no-stop-on-upgrade --no-start
 	dh_installinit --name=regenerate_ssh_host_keys --no-stop-on-upgrade --no-start
 	dh_installinit --name=wifi-country --no-stop-on-upgrade --no-start
-
-override_dh_systemd_enable:
-	dh_systemd_enable --name=rpi-display-backlight
-	dh_systemd_enable --name=sshswitch
-	dh_systemd_enable --name=apply_noobs_os_config --no-enable
-	dh_systemd_enable --name=regenerate_ssh_host_keys --no-enable
-	dh_systemd_enable --name=wifi-country
-
-override_dh_systemd_start:
-	dh_systemd_start --name=sshswitch --no-start
-	dh_systemd_start --name=apply_noobs_os_config --no-start
-	dh_systemd_start --name=regenerate_ssh_host_keys --no-start
-	dh_systemd_start --name=wifi-country --no-start
